File: DSA/JavaScript/Arrays/ReverseAnArray.js

function reverseArray(n, arr){
    let start = 0;
    let last = n -1;
    let temp;
    while(start<last){
        temp = arr[start];
        arr[start] = arr[last];
        arr[last] = temp;
        start++;
        last--;
    }
}


var myLink = document.getElementById('reverseAnArray');

myLink.onclick = function(){
    var n = window.prompt("Enter number of elements in array ");
    var arr=[];
    for(var i=0;i<n;i++){
        arr[i] = window.prompt("Enter ",i,"th element of array :");
    }
    reverseArray(n, arr);
    window.alert(arr)

}

File: DSA/C++/Arrays/ReverseAnArray.cpp
#include<iostream>
using namespace std;
void reverseArray(int n, int arr[]){
    int start = 0;
    int last = n -1;
    int temp;
    while(start<last){
        temp = arr[start];
        arr[start] = arr[last];
        arr[last] = temp;
        start++;
        last--;
    }
}
int main(){
    int n;
    cout<<"Enter number of elements in array: ";
    cin>>n;
    int arr[n];
    cout<<"\nEnter elements of array: ";
    for(int i=0;i<n;i++){
        cin>>arr[i];
    }
    reverseArray(n, arr);
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
    return 0;
}
